### PR TITLE
move defaultPrecisions to Params class

### PR DESCRIPTION
**Pull request by ndemashov**
### Details:
LPT has defaultPrecisions global non-const variable

### Tickets:
69448

### CODE REVIEW

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 1]
```diff
@@ -392,19 +392,21 @@ static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function>
     pass_config->enable<ngraph::pass::ConvertDetectionOutput1ToDetectionOutput8>();
 
     if (useLpt) {
+        const auto defaultPrecisions = ngraph::pass::low_precision::LowPrecision::getDefaultPrecisions(nGraphFunc);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/mkldnn_plugin.cpp:395]: Default precisions depend on FakeQuantize levels. Please, identify default precisions in one place upper and reuse it. Don't pass `getDefaultPrecisions` result. Default precisions identification you can find below:
```
        // for GNA networks reference execution
        bool updatePrecision = true;
        bool hasINT16orINT32Levels = ngraph::pass::low_precision::LowPrecision::isFQLevelsPresent(
                nGraphFunc,
                {levels::int16, levels::int16_narrow_range,
                 levels::int32, levels::int32_narrow_range});
        if (hasINT16orINT32Levels) {
            updatePrecision = false;
            LowPrecision::setDefaultPrecisions(precision_set::int8_int16_int32_support);

            supportedPrecisions = std::vector<OperationPrecisionRestriction>({});
        }
```

**Review by eshoguli — CHANGES_REQUESTED** [review]
Not totally completed yet, but PR requires changes

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp (hunk 1 above)
Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: done, thank you.

**Comment by eshoguli** [comment]
@vzinovie, can you help me review docs/snippets/lpt_mkldnn_plugin.cpp, please?

File: src/common/low_precision_transformations/include/low_precision/align_quantization_intervals.hpp [hunk 2]
```diff
@@ -30,5 +30,8 @@ class LP_TRANSFORMATIONS_API AlignQuantizationIntervals;
 class ngraph::pass::low_precision::AlignQuantizationIntervals : public ngraph::pass::FunctionPass {
 public:
     NGRAPH_RTTI_DECLARATION;
+    AlignQuantizationIntervals(const std::vector<ngraph::element::Type> defaultPrecisions = { ngraph::element::u8,  ngraph::element::i8 });
```
Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/align_quantization_intervals.hpp:33]: minor: `{ ngraph::element::u8,  ngraph::element::i8 }` => `ngraph::pass::low_precision::precision_set::int8_support`

Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/align_quantization_intervals.hpp:33]: minor: `{ ngraph::element::u8,  ngraph::element::i8 }` => `ngraph::pass::low_precision::precision_set::int8_support`

Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/align_quantization_intervals.hpp:33]: minor: `{ ngraph::element::u8,  ngraph::element::i8 }` => `ngraph::pass::low_precision::precision_set::int8_support`

Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/align_quantization_intervals.hpp:33]: minor: 
1. Use constant reference: `const std::vector<ngraph::element::Type> defaultPrecisions` => `const std::vector<ngraph::element::Type>& defaultPrecisions`
2. Use predefined member: `{ ngraph::element::u8,  ngraph::element::i8 }` => `ngraph::pass::low_precision::precision_set::int8_support`

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 3]
```diff
@@ -172,7 +172,7 @@ static void TransformationUpToCPUSpecificOpSet(const std::shared_ptr<ngraph::Fun
     manager.register_pass<ngraph::pass::InitNodeInfo>();
 
     const bool useLpt =
-            _enableLPT &&
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/mkldnn_plugin.cpp:175]: This alignment fix is absolute correct, but other updates are not connected LPT and can affect other team members. Can you, please, rollback alignment changes which is not connected with LPT? thanks

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 4]
```diff
@@ -415,20 +415,30 @@ static void TransformationUpToCPUSpecificOpSet(const std::shared_ptr<ngraph::Fun
...
+
     if (useLpt) {
+        bool hasINT16orINT32Levels = ngraph::pass::low_precision::LowPrecision::isFQLevelsPresent(
+            nGraphFunc,
+            { ngraph::pass::low_precision::levels::int16, ngraph::pass::low_precision::levels::int16_narrow_range,
+             ngraph::pass::low_precision::levels::int32, ngraph::pass::low_precision::levels::int32_narrow_range });
+        if (hasINT16orINT32Levels) {
+            defaultPrecisions = ngraph::pass::low_precision::precision_set::int8_int16_int32_support;
+        }
+
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/mkldnn_plugin.cpp:428]: As I noticed before this part has to be moved upper. `lpt_mkldnn_plugin.cpp` (which was updated) is partial copy of original `mkldnn_plugin.cpp` (which was not updated) to demonstrate code snippets in LPT documentation. Please, update `mkldnn_plugin.cpp` first and copy code snippets to `lpt_mkldnn_plugin.cpp` after.

**Review by eshoguli — CHANGES_REQUESTED** [review]
there are comments

File: src/common/low_precision_transformations/src/network_helper.cpp [hunk 5]
```diff
@@ -1292,7 +1299,10 @@ bool NetworkHelper::isQuantizeSupported(const std::shared_ptr<opset1::FakeQuanti
     return QuantizationDetails::outputLayoutIsSupported(fakeQuantize) && QuantizationDetails::isSupportedLevel(fakeQuantize->get_levels());
 }
 
-FakeQuantizeDequantization NetworkHelper::getDequantization(const std::shared_ptr<const Node>& node, const size_t parentIndex, const bool inPlace) {
+FakeQuantizeDequantization NetworkHelper::getDequantization(const std::shared_ptr<const Node>& node,
+    const std::vector<ngraph::element::Type> _defaultPrecisions,
```
Comment (eshoguli) [inline src/common/low_precision_transformations/src/network_helper.cpp:1303]: `_defaultPrecisions` => `defaultPrecisions`

File: src/common/low_precision_transformations/include/low_precision/align_quantization_parameters.hpp [hunk 6]
```diff
@@ -31,5 +31,8 @@ class LP_TRANSFORMATIONS_API AlignQuantizationParameters;
 class ngraph::pass::low_precision::AlignQuantizationParameters : public ngraph::pass::FunctionPass {
 public:
     NGRAPH_RTTI_DECLARATION;
+    AlignQuantizationParameters(const std::vector<ngraph::element::Type> defaultPrecisions = { ngraph::element::u8, ngraph::element::i8 });
```
[the remark this answers is not in the document]
Comment (ndemashov) [reply src/common/low_precision_transformations/include/low_precision/align_quantization_parameters.hpp:34]: done, thank you.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp (hunk 4 above)
Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: done, thank you.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 7]
```diff
@@ -423,12 +429,12 @@ static void TransformationUpToCPUSpecificOpSet(const std::shared_ptr<ngraph::Fun
             return !MKLDNNFakeQuantizeNode::isSupportedOperation(node, errMsg);
         });
 
-        pass_config->set_callback<ngraph::pass::ConvertQuantizeDequantize>([](const_node_ptr &node) -> bool {
-            return ngraph::pass::low_precision::NetworkHelper::areQuantizeAndDequantizeSupportedForMultiply(node);
+        pass_config->set_callback<ngraph::pass::ConvertQuantizeDequantize>([defaultPrecisions](const_node_ptr &node) -> bool {
```
Comment (vzinovie) [inline src/plugins/intel_cpu/src/mkldnn_plugin.cpp:432]: Capture `defaultPrecisions` by reference
```suggestion
        pass_config->set_callback<ngraph::pass::ConvertQuantizeDequantize>([&defaultPrecisions](const_node_ptr &node) -> bool {
```

**Review by vzinovie — CHANGES_REQUESTED** [review]
Reviewed mkldnn_plugin.cpp only

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp (hunk 3 above)
Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp:175]: done, thank you.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp (hunk 4 above)
Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: done, thank you.

File: src/common/low_precision_transformations/src/align_quantization_parameters.cpp [hunk 8]
```diff
@@ -9,19 +9,23 @@
...
 bool ngraph::pass::low_precision::AlignQuantizationParameters::run_on_model(const std::shared_ptr<ngraph::Function>& f) {
     ngraph::pass::Manager manager;
     manager.set_per_pass_validation(false);
     std::shared_ptr<ngraph::pass::GraphRewrite> propagation = manager.register_pass<ngraph::pass::GraphRewrite>();
-    propagation->add_matcher<low_precision::CreateAttribute<QuantizationAlignmentAttribute>>();
-    propagation->add_matcher<low_precision::PropagateThroughPrecisionPreserved<QuantizationAlignmentAttribute>>();
-    propagation->add_matcher<low_precision::UpdateSharedPrecisionPreserved<QuantizationAlignmentAttribute, PerTensorQuantizationAttribute>>();
+    propagation->add_matcher<low_precision::CreateAttribute<QuantizationAlignmentAttribute>>(AttributeParameters());
+    propagation->add_matcher<low_precision::PropagateThroughPrecisionPreserved<QuantizationAlignmentAttribute>>(defaultPrecisions);
+    propagation->add_matcher<low_precision::UpdateSharedPrecisionPreserved<QuantizationAlignmentAttribute, PerTensorQuantizationAttribute>>(defaultPrecisions);
```
[the remark this answers is not in the document]
Comment (ndemashov) [reply src/common/low_precision_transformations/src/align_quantization_parameters.cpp]: done, thank you.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp (hunk 7 above)
Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: done, thank you.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 9]
```diff
@@ -174,9 +174,16 @@ static void TransformationUpToCPUSpecificOpSet(const std::shared_ptr<ngraph::Fun
     const bool useLpt =
             _enableLPT &&
         ngraph::pass::low_precision::LowPrecision::isFunctionQuantized(nGraphFunc);
+    auto defaultPrecisions = ngraph::pass::low_precision::precision_set::int8_support;
     if (useLpt) {
-        manager.register_pass<ngraph::pass::DisableConvertConstantFoldingOnConstPath>(
-            std::vector<ngraph::element::Type>{ ngraph::element::i8, ngraph::element::u8, ngraph::element::i4, ngraph::element::u4 });
+        bool hasINT16orINT32Levels = ngraph::pass::low_precision::LowPrecision::isFQLevelsPresent(
```
[the remark this answers is not in the document]
Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: done, thank you.

File: src/common/low_precision_transformations/src/network_helper.cpp (hunk 5 above)
Comment (ndemashov) [reply src/common/low_precision_transformations/src/network_helper.cpp]: done, thank you.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 10]
```diff
@@ -174,9 +174,16 @@ static void TransformationUpToCPUSpecificOpSet(const std::shared_ptr<ngraph::Fun
     const bool useLpt =
             _enableLPT &&
         ngraph::pass::low_precision::LowPrecision::isFunctionQuantized(nGraphFunc);
+    auto defaultPrecisions = ngraph::pass::low_precision::precision_set::int8_support;
+    bool hasINT16orINT32Levels = ngraph::pass::low_precision::LowPrecision::isFQLevelsPresent(
```
Comment (vzinovie) [inline src/plugins/intel_cpu/src/mkldnn_plugin.cpp:178]: Just initialize as `false` or `true` and call the function inside `if (useLpt)`.

Comment (ndemashov) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: Didn't understand at first. Thanks for the clarification. It's done

**Review by eshoguli — APPROVED** [review]

File: src/common/low_precision_transformations/include/low_precision/propagate_through_precision_preserved.hpp [hunk 11]
```diff
@@ -99,11 +99,11 @@ class ngraph::pass::low_precision::PropagateThroughPrecisionPreserved : public n
...
 
     // TODO: possible duplicate: PropagateToInput::getSourceOutputAttribute
-    std::vector<ov::Any> getParentInputRestrictions(
-        const std::shared_ptr<ngraph::Node> node) {
+    std::vector<ov::Any> getParentInputRestrictions(const std::shared_ptr<ngraph::Node> node,
+        const std::vector<ngraph::element::Type>& defaultPrecisions) {
         std::vector<ov::Any> parentAttributes;
-        auto getInput = [](const std::shared_ptr<ngraph::Node>& node, const size_t index) -> Input<Node> {
-            const auto dequantization = NetworkHelper::getDequantization(node, index);
+        auto getInput = [defaultPrecisions](const std::shared_ptr<ngraph::Node>& node, const size_t index) -> Input<Node> {
```
Comment (vzinovie) [inline src/common/low_precision_transformations/include/low_precision/propagate_through_precision_preserved.hpp:105]: Could you please make all lambda captures "by ref" if copy is not required by lambda.

**Review by vzinovie — CHANGES_REQUESTED** [review]

File: src/common/low_precision_transformations/include/low_precision/propagate_through_precision_preserved.hpp (hunk 11 above)
Comment (ndemashov) [reply src/common/low_precision_transformations/include/low_precision/propagate_through_precision_preserved.hpp]: All lambda captures are made by ref. Thank you

**Review by vzinovie — APPROVED** [review]

**Pull request merged**